psi4/driver/p4util/python_helpers.py

```diff
@@ -71,16 +71,15 @@ def _resolve_target(key, target):
     # if a string, they search for a gbs file with that name.
     # if a function, it needs to apply a basis to each atom.
 
-    basisdict = qcdb.BasisSet.pyconstruct(
-        mol.create_psi4_string_from_molecule(), key, resolved_target, fitrole, other, return_atomlist=return_atomlist)
+    basisdict = qcdb.BasisSet.pyconstruct(mol.to_dict(),
+                                          key, resolved_target, fitrole, other, return_atomlist=return_atomlist)
```

Comment:
Hmm, I think we can overload the `__dict__` attribute so that `dict(mol)` will work as well.

Reply:
Would that work on `psi4.core.Molecule`, too, though?

Reply:
Yea, since everything is a dict all of the special Python functions like `dict`, `+=`, etc just look for a object that has the right attributes. Kinda handy.

Reply:
Still makes me nervous b/c for both Mol classes, the to_dict could be not a full picture to recreate (say, Zmat or xyz with vars). Result of `to_dict` is a fine, upstanding representation of the Mol and can be reversed with `from_dict`, but it's not necessarily the same repr of the Mol instance.